### PR TITLE
Add DG - Appendix: Effort

### LINKED ISSUE #450: Appendix: Effort

### PR DESCRIPTION
**Pull request by zhonghan721**
Closes #450

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -1894,3 +1894,41 @@ testers are expected to do more _exploratory_ testing.
...
+
+### Difficulty level
+
+This project was moderately challenging as we had to deal with a large existing code base.
+It took us some time to understand how the different components interact with each other.
+Furthermore, our team was not familiar with frameworks such as JavaFX prior to this, and we had to learn how to use it.
+
+<br>
+
+### Challenges faced
```
Comment (jianyangg) [inline docs/DeveloperGuide.md:1912]: Sounds good, but perhaps we could add the challenge of figuring out how to hide the app behind a login / logout system where the password is hashed and how this account data has to be stored? Something like that haha

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -1894,3 +1894,41 @@ testers are expected to do more _exploratory_ testing.
...
+    Furthermore, there is a dependency between the two entities, requiring a new `BookStorageWithReference` 
+    class that accepts two type parameters to be created.
+* Adapting the `PersonListPanel` to `ListPanel`
+  * The `PersonListPanel` was designed to contain a list of `Person`. However, as we decided to display both 
+    `Customer` and `Delivery` in the same list, we had to adapt the `PersonListPanel` to `ListPanel` to 
+    accommodate both types of entities.
+
+<br>
+
+### Achievements
```
Comment (jianyangg) [inline docs/DeveloperGuide.md:1931]: Perhaps: Added security feature to prevent unauthorised access to the application.

**Review by jianyangg — COMMENTED** [review]
Minor comments, otherwise LGTM.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (zhonghan721) [reply docs/DeveloperGuide.md:1912]: Added!

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (zhonghan721) [reply docs/DeveloperGuide.md:1931]: Added!

**Review by juliusgambe — APPROVED** [review]
LGTM

**Pull request merged**